test/framework/integration_framework/integration_test_framework.cpp

```diff
@@ -269,27 +269,26 @@ namespace integration_framework {
   };
 
   IntegrationTestFramework &IntegrationTestFramework::sendTx(
-      const shared_model::proto::Transaction &tx, std::set<TxStatus> statuses) {
+      const shared_model::proto::Transaction &tx,
+      std::function<
+          void(const std::vector<shared_model::proto::TransactionResponse> &)>
+          validation,
+      size_t status_count) {
     std::condition_variable cv;
     std::mutex m;
+    std::vector<shared_model::proto::TransactionResponse> status_list;
 
     // statuses number that haven't being appeared yet
-    size_t count = statuses.size();
     iroha_instance_->instance_->getStatusBus()
         ->statuses()
         .filter([&](auto s) { return s->transactionHash() == tx.hash(); })
-        .map([](auto s) {
-          return boost::apply_visitor(TxStatusVisitor(), s->get());
-        })
-        .filter([&statuses](auto s) { return statuses.count(s) != 0; })
-        .take_while([&statuses](auto) { return statuses.size() != 0; })
+        .take_while([&status_count](auto) { return status_count != 0; })
         .subscribe([&](auto s) {
-          auto response = statuses.find(s);
-          if (response != statuses.end()) {
-            statuses.erase(response);
-          }
+          status_list.emplace_back(iroha::protocol::ToriiResponse(
```

Comment:
Can we make status list to be vector of interfaces and use clone function instead of copying protobuf explicitly?